pkg/install/install.go

```diff
@@ -100,12 +101,51 @@ func DefaultImage() string {
 	return image
 }
 
+func validMailAddress(address string) (string, bool) {
+	addr, err := mail.ParseAddress(address)
+	if err != nil {
+		return "", false
+	}
+	return addr.Address, true
+}
+
 func Install(ctx context.Context, image string, opts *Options) error {
 	// I don't want these errors on the screen. Probably a better way to do this.
 	klog.SetOutput(io.Discard)
 	klogv2.SetOutput(io.Discard)
 	utilruntime.ErrorHandlers = nil
 
+	// Require E-Mail address when using Let's Encrypt production
+	if opts.Config.LetsEncrypt != nil && *opts.Config.LetsEncrypt == "enabled" {
+		if opts.Config.LetsEncryptTOSAgree == nil || !*opts.Config.LetsEncryptTOSAgree {
+			ok, err := pterm.DefaultInteractiveConfirm.Show("You are choosing to use Let's Encrypt production environment for TLS certificates. To do so, you must agree to their Terms of Service: https://community.letsencrypt.org/tos\nTip: use --lets-encrypt-tos-agree to skip this prompt\nDo you agree to Let's Encrypt TOS?")
+			if err != nil {
+				return err
+			}
+			if !ok {
+				return fmt.Errorf("you must agree to Let's Encrypt TOS to use their production environment")
+			}
+			opts.Config.LetsEncryptTOSAgree = &ok
+		}
+		if opts.Config.LetsEncryptEmail == "" {
+			result, err := pterm.DefaultInteractiveTextInput.WithMultiLine(false).Show("Enter your email address for Let's Encrypt (required for production)")
+			if err != nil {
+				return err
+			}
+			opts.Config.LetsEncryptEmail = result
+		}
+		pterm.Info.Println("You've enabled automatic TLS certificate provisioning with Let's Encrypt production. This can take a few minutes to configure.")
+	}
+
+	// Validate E-Mail address provided for Let's Encrypt registration
+	if opts.Config.LetsEncryptEmail != "" {
```

Comment:
Found a bug with this: If I use the prompts to enter my email, it lets me just hit enter to submit a blank email.
That lets the install continue with a blank email and then the controller gets in an error loop.

So, this check probably needs to add an "or" for when letsEncrypt is enabled, like `if opts.Config.LetsEncryptEmail != "" || opts.COnfig.LetsEncrypt == "enabled {`

Reply:
Good catch! Alternatively we could make `LetsEncryptEmail` a `*string` type flag as well and check for `nil` instead of `""`.
Went for or check for now :heavy_check_mark: